File: 백준/Bronze/1157. 단어 공부/단어 공부.py
word = input()
word = word.upper()
words = {}
count = 0

for i in word:
    if i not in words:
        words[i] = 1
    else:
        words[i] += 1

new_list = [i for i in words.values()]

for i in new_list:
    if max(new_list) == i:
        count += 1

if count == 1:
    for i in words.keys():
        if words[i] == max(new_list):
            print(i)
else:
    print("?")

File: 백준/Bronze/1157. 단어 공부/단어 공부.c
#include <stdio.h>
#include <string.h>

char str[1000003];

int main() {

	int alphabet[26] = { 0, };
	int max, count = 0;
	scanf("%s", str);
    int len = strlen(str);

	for (int i = 0; i < len; i++) {
		if (str[i] >= 'A' && str[i] <= 'Z') {
			alphabet[str[i] - 'A']++;
		}
		else {
			alphabet[str[i] - 'a']++;
		}
	}

	max = 0;
	for (int i = 1; i < 26; i++) {
		if (alphabet[max] < alphabet[i]) {
			max = i;
		}
	}

	for (int i = 0; i < 26; i++) {
		if (alphabet[max] == alphabet[i]) {
			count++;
		}
	}

	if (count > 1) {
		printf("?");
	}
	else {
		printf("%c", max + 'A');
	}

}